hedera-node/src/main/java/com/hedera/services/ServicesState.java

```diff
@@ -164,14 +167,22 @@ public void addDeserializedChildren(List<MerkleNode> children, int version) {
 	@Override
 	public void migrate() {
 		int deserializedVersionFromState = getDeserializedVersion();
+		boolean migrationProcessed = false;
 		if (deserializedVersionFromState < RELEASE_0240_VERSION) {
 			stakeFundingMigrator.accept(this);
 			if (deserializedVersionFromState < RELEASE_0220_VERSION) {
 				// Remove after confirming no regression tests use 0.21.x states
-				blobMigrator.migrateFromBinaryObjectStore(this, deserializedVersionFromState);
-				init(getPlatformForDeferredInit(), getAddressBookForDeferredInit(), getDualStateForDeferredInit());
+				blobMigrator.migrate(this, deserializedVersionFromState);
+				migrationProcessed = true;
```

Comment:
The curious-looking "deferred `init()`" pattern is unnecessary; the `0.22` migration touched data stored in PostgreSQL and that's why it was required then.

Reply:
If I don't defer the init() pattern, then I get a "ERROR: Signed state loaded from disk has an invalid hash!"